FrejaEidClient/src/main/java/com/verisec/frejaeid/client/http/HttpService.java

```diff
@@ -168,6 +174,63 @@ public final <Response extends FrejaHttpResponse> Response send(String methodUrl
         }
     }
 
+    @Override
+    public final HttpResponse httpGet(String methodUrl, Map<String, String> parameters)
+            throws FrejaEidClientInternalException, FrejaEidException, UnsupportedEncodingException {
+
+        HttpResponse httpResponse = null;
+        HttpGet request = null;
+        HttpStatusCode httpStatusCode = null;
+
+        URIBuilder uriBuilder = new URIBuilder();
+        uriBuilder.setPath(methodUrl);
+        for (Map.Entry<String, String> parameter : parameters.entrySet()) {
+            uriBuilder.addParameter(parameter.getKey(), URLEncoder.encode(parameter.getValue(),
+                                                                          StandardCharsets.UTF_8.toString()));
+        }
+
+        try {
+            request = new HttpGet(uriBuilder.build());
+            request.addHeader("Content-Type", "application/json");
+            request.addHeader(HttpHeaders.USER_AGENT, userAgentHeader);
+            httpResponse = httpClient.execute(request);
+            LOG.debug("Successfully sent HttpGet request to address {}.", methodUrl);
+            int httpStatusCodeValue = httpResponse.getStatusLine().getStatusCode();
+            String responseString =  EntityUtils.toString(httpResponse.getEntity(), StandardCharsets.UTF_8);
+            httpStatusCode = HttpStatusCode.getHttpStatusCode(httpStatusCodeValue);
+            if (httpStatusCode == null) {
+                throw new FrejaEidException(
+                        String.format("Received unsupported HTTP status code %s. Received HTTP message: %s.",
+                                      httpResponse.getStatusLine().getStatusCode(), responseString));
+            }
+            switch (httpStatusCode) {
+                case OK:
+                case NO_CONTENT:
+                    return httpResponse;
+                case BAD_REQUEST:
+                case UNPROCESSABLE_ENTITY:
+                    FrejaHttpErrorResponse errorResponse =
```

Comment:
In a case of Freja requests we can receive an error response from RP service that we can deserialize, but here an error can happen only if something went wrong like wrong request, resource service not working etc... So, here we can just return an exception in case of any error.

Reply:
Done.